include/deal.II/fe/mapping_manifold.h

```diff
@@ -0,0 +1,581 @@
+   * Mapping::InternalDataBase for an extensive description.
+   *
+   * For the current class, the InternalData class stores data that is
+   * computed once when the object is created (in get_data()) as well as data
+   * the class wants to store from between the call to fill_fe_values(),
+   * fill_fe_face_values(), or fill_fe_subface_values() until possible later
+   * calls from the finite element to functions such as transform(). The
+   * latter class of member variables are marked as 'mutable'.
+   */
+  class InternalData : public Mapping<dim,spacedim>::InternalDataBase
+  {
+  public:
+    /**
+     * Constructor.
+     */
+    InternalData();
+
+    /**
+     * Initialize the object's member variables related to cell data based on
+     * the given arguments.
+     *
+     * The function also calls compute_shape_function_values() to actually set
+     * the member variables related to the values and derivatives of the
+     * mapping shape functions.
+     */
+    void
+    initialize (const UpdateFlags      update_flags,
+                const Quadrature<dim> &quadrature,
+                const unsigned int     n_original_q_points);
+
+    /**
+     * Initialize the object's member variables related to cell and face data
+     * based on the given arguments. In order to initialize cell data, this
+     * function calls initialize().
+     */
+    void
+    initialize_face (const UpdateFlags      update_flags,
+                     const Quadrature<dim> &quadrature,
+                     const unsigned int     n_original_q_points);
+
+
+    /**
+     * Compute the weights associated to the Manifold object, that
+     * need to be passed when computing the location of the quadrature
+     * points.
+     */
+    void
+    compute_manifold_quadrature_weights(const Quadrature<dim> &quadrature);
+
+    /**
+     * Store vertices internally.
+     */
+    void
+    store_vertices(const typename Triangulation<dim,spacedim>::cell_iterator &cell) const;
+
+    /**
+     * Return an estimate (in bytes) or the memory consumption of this object.
+     */
+    virtual std::size_t memory_consumption () const;
+
+    /**
+     * The current cell vertices.
+     *
+     * Computed each.
+     */
+    mutable std::vector<Point<spacedim> > vertices;
+
+    /**
+     * The current cell.
+     *
+     * Computed each.
+     */
+    mutable typename Triangulation<dim,spacedim>::cell_iterator cell;
+
+    /**
+     * The actual quadrature on the reference cell.
+     *
+     * Computed once.
+     */
+    Quadrature<dim> quad;
+
+
+    /**
+     * Values of quadrature weights for manifold quadrature formulas.
+     *
+     * Computed once.
+     */
+    std::vector<std::vector<double> > cell_manifold_quadrature_weights;
+
+    // /**
+    //  * Values of shape function derivatives. Access by function @p derivative.
+    //  *
+    //  * Computed once.
+    //  */
+    // std::vector<Tensor<1,dim> > shape_derivatives;
```

Comment:
I think you don't need these arrays any more, right? Remove what you don't need!